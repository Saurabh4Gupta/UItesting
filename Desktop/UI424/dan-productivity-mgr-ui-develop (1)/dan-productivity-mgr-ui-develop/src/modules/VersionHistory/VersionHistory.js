import React, { useState, useEffect } from 'react';
import { useLazyQuery } from '@apollo/client';
import PropTypes from 'prop-types';
import {
  FileLockup,
  Layout,
  IconButton,
  List,
  Card,
  Button,
  Stack,
} from '@dentsu-ui/components';
import { slice, concat } from 'lodash';
import { dataFieldCms as PageContent } from '../../cms';
import { DOWNLOAD_FILE } from '../ViewProdDataRequest/queries';

const VersionHistory = (props) => {
  const { trackerFiles, trackerTemplate } = props;
  const [urlData, setUrlData] = useState();
  const [blobData, setBlobData] = useState();
  const [getDownloadFile, { data }] = useLazyQuery(DOWNLOAD_FILE);

  useEffect(() => {
    getDownloadFile({ variables: { id: blobData } })
    if (data) {
      setUrlData(data.downloadFile.data)
    }
  }, [blobData, data]);

  const reversedData = trackerFiles.length
    ? [...trackerFiles].reverse()
    : [
        {
          id: 1,
          name: trackerTemplate[0].name,
          type: 'xls',
          size: '242KB',
          version: 'Template',
          dataRequest: '',
          blobId: trackerTemplate[0].blobId,
          createdAt: '2020-12-01',
        },
      ];
  const len = reversedData.length;
  const lattestVersion = reversedData[0].version;

  const LIMIT = 10;
  const [list, setList] = useState(slice(reversedData, 0, LIMIT));
  const [index, setIndex] = useState(LIMIT);

  const loadMore = () => {
    const newIndex = index + LIMIT;
    const newList = concat(list, slice(reversedData, index, newIndex));
    setIndex(newIndex);
    setList(newList);
  };

  const downloadButton = (blobId, name) => {
    setBlobData(blobId);
    return (
      <>
        <a
          href={urlData}
          download={name}
          style={{ color: 'black' }}
        >
          <IconButton variant="ghost" icon="download" size="small" />
        </a>
      </>
    )
  }
    //

  return (
    <>
      <Layout>
        <Layout.Section size="2/3">
          <Layout.Panel>
            {/* <FileLockup name="Microsoft UK " extension="xls" filesize={123450000} hasStatus statusLabel="Current version" action={menu} /> */}
          </Layout.Panel>
        </Layout.Section>
        <Layout.Section size="1/3">
          <Layout.Panel>
            <Card.Title>{PageContent.versionHistoryTitle}</Card.Title>

            <List
              hasTotal={false}
              isSearchable={false}
              items={list}
              total={reversedData.length}
              renderItem={(item) => {
                const { name, size, type, version, blobId } = item;
                return (
                  <FileLockup
                    style={{ padding: '20px' }}
                    name={`${name} ${version}`}
                    extension={type}
                    filesize={size}
                    hasStatus={version === lattestVersion}
                    statusLabel={
                      version === lattestVersion
                        ? PageContent.currentVersionText
                        : ''
                    }
                    statusVariant="positive"
                    action={downloadButton(blobId, name)}
                    size="medium"
                  />
                );
              }}
            />

            {len >= 10 && list.length < len ? (
              <Stack justifyContent="center" style={{ marginTop: '25px' }}>
                <Button
                  variant="ghost"
                  size="small"
                  iconRight="refresh"
                  style={{ color: 'black' }}
                  onClick={loadMore}
                >
                  {PageContent.loadMoreText}
                </Button>
              </Stack>
            ) : (
              false
            )}
          </Layout.Panel>
        </Layout.Section>
      </Layout>
    </>
  );
};
VersionHistory.propTypes = {
  trackerFiles: PropTypes.arrayOf(PropTypes.object),
  trackerTemplate: PropTypes.arrayOf(PropTypes.object),
};
VersionHistory.defaultProps = {
  trackerTemplate: PropTypes.arrayOf(PropTypes.object),
  trackerFiles: [],
};

export default VersionHistory;
