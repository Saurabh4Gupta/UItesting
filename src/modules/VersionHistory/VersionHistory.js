/* eslint-disable radix */
import React, { useState } from 'react';
import { List, Link, Image, Stack, Icon, Divider, Chip, Button } from '@dentsu-ui/components';
import Box from '@dentsu-ui/components/dist/cjs/components/Box';
import Card from '@dentsu-ui/components/dist/cjs/components/Card';
import {
  slice, concat,
} from 'lodash';
import { versionHistory } from '../Mock/mockData';
// import { dataFieldCms as PageContent } from '../../cms';


const VersionHistory = () => {
  const len = versionHistory.data.length;
  const LIMIT = 10;
  const [showMore, setShowMore] = useState(true);
  const [list, setList] = useState(slice(versionHistory.data, 0, LIMIT));
  const [index, setIndex] = useState(LIMIT);

  const loadMore = () => {
    const newIndex = index + LIMIT;
    const newShowMore = newIndex < (len - 1);
    const newList = concat(list, slice(versionHistory.data, index, newIndex));
    setIndex(newIndex);
    setList(newList);
    setShowMore(newShowMore);
    // console.log('newIndex', newIndex);
  }
  // console.log('listdata', list)
  // console.log('Index', index);
  console.log(showMore)
  return (
    <>
      <Box ml="60%" mr="65px" mt="20px" mb="100px">
        <Card>
          <Card.Title>
            Version history
          </Card.Title>
          <List
            hasTotal={false}
            isSearchable={false}
            items={list}
            total={versionHistory.data.length}
            renderItem={(item) => {
              const { name, size, type, version } = item;
              const dataLength = versionHistory.data.length;
              return (
                <Link url="/DataMappings.xlsx" style={{ color: 'black' }}>
                  <Stack
                    flexDirection="row"
                    justifyContent="space-between"
                    style={{ border: '1px solid #f6f6f6' }}
                    mb="2"
                    width="370px"
                    mt="2"

                  >
                    <Box mt="12px" ml="12px">
                      <Card.Thumbnail>
                        <Image
                          htmlWidth={40}
                          height={40}
                          objectFit="contain"
                          src="/microsoft-excel-icon.png"
                          fallbackSrc="https://via.placeholder.com/150"
                        />
                      </Card.Thumbnail>
                    </Box>

                    <Stack flexDirection="column" mb="20px">
                      <Box style={{ fontSize: '12px' }} mt="15px" ml="10px">
                        <b>
                          {name}
                          {version}
                        </b>
                      </Box>
                      <Box style={{ fontSize: '12px' }} ml="10px" mt="7px">
                        {size}
                        <Divider orientation="vertical" />
                        {type}
                        {dataLength > 1 ? (
                          <>
                            {version === `V${dataLength - 1}.0` ? (
                              <>
                                <Divider orientation="vertical" />
                                <Chip variant="status" status="neutral" style={{ fontSize: '8px' }}>
                                  current version
                                </Chip>
                              </>
                            ) : false}
                          </>
                        ) : (
                          <>
                            <Divider orientation="vertical" />
                            <Chip variant="status" status="neutral" style={{ fontSize: '8px' }}>
                              current version
                            </Chip>
                          </>
                          )}
                      </Box>
                    </Stack>
                    <Box>
                      <Icon
                        style={{ marginTop: '15px', marginRight: '10px', marginLeft: '25px' }}
                        icon="download"
                        color="blue"
                        size="14"
                      />
                    </Box>
                  </Stack>
                </Link>
              );
            }}


          />
          {len >= 10 ? (
            <Box style={{ margin: '10px auto' }}>
              <Button variant="ghost" size="small" iconRight="refresh" style={{ color: 'black' }} onClick={loadMore}>
                Load more

              </Button>
            </Box>

          ) : false}


        </Card>

      </Box>
    </>
  )
}

export default VersionHistory;
