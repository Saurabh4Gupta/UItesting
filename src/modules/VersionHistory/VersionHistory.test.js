import React from 'react';
import { shallow } from 'enzyme';
import VersionHistory from './VersionHistory';

describe('test cases for Table list', () => {
  let wrapper;
  const list = {
    id: 1,
    name: 'Microsoft Uk - Productivity Report - ',
    type: 'XLS',
    size: '242KB',
    version: 'Template',
    dataRequest: '',
    blobId: '',
    createdAt: '2020-12-01',
  };
  beforeAll(() => {
    wrapper = shallow(<VersionHistory />);
  });
  it('should match snapshot', () => {
    wrapper.find('List').props().renderItem(list);
    expect(wrapper).toMatchSnapshot();
  });
  it('Test loadmore click event', () => {
    wrapper.find({ children: 'Load more' }).simulate('click');
  });
});
