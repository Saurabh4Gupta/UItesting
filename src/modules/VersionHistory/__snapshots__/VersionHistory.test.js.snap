// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test cases for Table list should match snapshot 1`] = `
<ForwardRef>
  <LayoutSection>
    <ForwardRef />
  </LayoutSection>
  <LayoutSection
    size="1/3"
  >
    <ForwardRef>
      <ForwardRef>
        Version history
      </ForwardRef>
      <List
        hasTotal={false}
        isSearchable={false}
        items={
          Array [
            Object {
              "blobId": "",
              "createdAt": "2020-12-01",
              "dataRequest": "",
              "id": 13,
              "name": "Microsoft Uk - Productivity Report - ",
              "size": "242KB",
              "type": "XLS",
              "version": "V12.0",
            },
            Object {
              "blobId": "",
              "createdAt": "2020-12-01",
              "dataRequest": "",
              "id": 12,
              "name": "Microsoft Uk - Productivity Report - ",
              "size": "242KB",
              "type": "XLS",
              "version": "V11.0",
            },
            Object {
              "blobId": "",
              "createdAt": "2020-12-01",
              "dataRequest": "",
              "id": 11,
              "name": "Microsoft Uk - Productivity Report - ",
              "size": "242KB",
              "type": "XLS",
              "version": "V10.0",
            },
            Object {
              "blobId": "",
              "createdAt": "2020-12-01",
              "dataRequest": "",
              "id": 10,
              "name": "Microsoft Uk - Productivity Report - ",
              "size": "242KB",
              "type": "XLS",
              "version": "V9.0",
            },
            Object {
              "blobId": "",
              "createdAt": "2020-12-01",
              "dataRequest": "",
              "id": 9,
              "name": "Microsoft Uk - Productivity Report - ",
              "size": "242KB",
              "type": "XLS",
              "version": "V8.0",
            },
            Object {
              "blobId": "",
              "createdAt": "2020-12-01",
              "dataRequest": "",
              "id": 8,
              "name": "Microsoft Uk - Productivity Report - ",
              "size": "242KB",
              "type": "XLS",
              "version": "V7.0",
            },
            Object {
              "blobId": "",
              "createdAt": "2020-12-01",
              "dataRequest": "",
              "id": 7,
              "name": "Microsoft Uk - Productivity Report - ",
              "size": "242KB",
              "type": "XLS",
              "version": "V6.0",
            },
            Object {
              "blobId": "",
              "createdAt": "2020-12-01",
              "dataRequest": "",
              "id": 6,
              "name": "Microsoft Uk - Productivity Report - ",
              "size": "242KB",
              "type": "XLS",
              "version": "V5.0",
            },
            Object {
              "blobId": "",
              "createdAt": "2020-12-01",
              "dataRequest": "",
              "id": 5,
              "name": "Microsoft Uk - Productivity Report - ",
              "size": "242KB",
              "type": "XLS",
              "version": "V4.0",
            },
            Object {
              "blobId": "",
              "createdAt": "2020-12-01",
              "dataRequest": "",
              "id": 4,
              "name": "Microsoft Uk - Productivity Report - ",
              "size": "242KB",
              "type": "XLS",
              "version": "V3.0",
            },
          ]
        }
        renderItem={[Function]}
        total={13}
      />
      <Box
        style={
          Object {
            "margin": "10px auto",
          }
        }
      >
        <ForwardRef
          iconRight="refresh"
          onClick={[Function]}
          size="small"
          style={
            Object {
              "color": "black",
            }
          }
          variant="ghost"
        >
          Load more
        </ForwardRef>
      </Box>
    </ForwardRef>
  </LayoutSection>
</ForwardRef>
`;
